datafusion/core/src/physical_planner.rs

```diff
@@ -1797,11 +1797,19 @@ impl DefaultPhysicalPlanner {
                             Err(e) => return Err(e),
                         }
                     }
-                    Err(e) => stringified_plans
-                        .push(StringifiedPlan::new(InitialPhysicalPlan, e.to_string())),
```

Comment:
I played around with this some more this morning -- I think the reason the error isn't displayed is that `InitialPhysicalPlan` is not shown by default in explain plans.